### PR TITLE
feat: Ask User for call quality feedback [WPB-9452]

### PR DESCRIPTION
**Pull request by borichellow**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-9452" title="WPB-9452" target="_blank">TaskWPB-9452</a>  [Android] : Call Quality Feedback - First Pass
  </td></table>
  <br />
 

# What's new in this PR?

### Issues

Every at least 3 days, when user ended a call we want to ask Call Quality feedback.

### Causes (Optional)

There is a task saying so.

### Solutions

Implement it, by adding `ShouldAskCallFeedbackUseCase` which will be called on every call ending and if user should asked for feedback kalium will call updated `EndCallResultListener` 
And new `ObserveAskCallFeedbackUseCase` which observes flow `EndCallResultListener` for corresponding result when user should or should not be asked for feedback. 

### Not related changes: 

while updating tests I noticed that we have 2 `CallRepositoryArrangement`, one in `arrangement.repository` one in `arrangement` package. So i merge them into one.

### CODE REVIEW

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/user/ShouldAskCallFeedbackUseCase.kt [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+@Suppress("FunctionNaming")
+internal fun ShouldAskCallFeedbackUseCase(
+    userConfigRepository: UserConfigRepository
+) = object : ShouldAskCallFeedbackUseCase {
+
+    override suspend fun invoke(): Boolean =
+        userConfigRepository.getNextTimeForCallFeedback().map {
+            it > 0L && DateTimeUtil.currentInstant() > Instant.fromEpochMilliseconds(it)
+        }.getOrElse(true)
```
Comment (alexandreferris) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/user/ShouldAskCallFeedbackUseCase.kt:44]: the `.getOrElse(true)` will mean that the first call of a fresh logged in user will display the feedback right?

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/call/usecase/EndCallResultListener.kt [hunk 2]
```diff
@@ -17,25 +17,34 @@
...
     }
+
+    override suspend fun onCallEndedAskForFeedback(shouldAsk: Boolean) {
+        conversationCallEnded.emit(EndCallResult.AskForFeedback(shouldAsk))
+    }
+}
+
+sealed class EndCallResult {
+    data object VerificationDegraded : EndCallResult()
+    data class AskForFeedback(val shouldAsk: Boolean) : EndCallResult()
```
Comment (alexandreferris) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/call/usecase/EndCallResultListener.kt:49]: 😍

**Review by alexandreferris — APPROVED** [review]
Great work!

Just have a tiny question just to make sure I understood whats expected, but approved nonetheless 👌🏻

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/call/usecase/EndCallUseCase.kt [hunk 3]
```diff
@@ -46,6 +47,8 @@ interface EndCallUseCase {
 internal class EndCallUseCaseImpl(
     private val callManager: Lazy<CallManager>,
     private val callRepository: CallRepository,
+    private val endCallListener: EndCallResultListener,
+    private val shouldAskCallFeedbackUseCase: ShouldAskCallFeedbackUseCase,
```
Comment (yamilmedina) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/call/usecase/EndCallUseCase.kt:51]:
```suggestion
    private val shouldAskCallFeedback: ShouldAskCallFeedbackUseCase,
```

**Review by yamilmedina — APPROVED** [review]
Just a small naming convention nitpick. great work 🚀

**Pull request merged**